eng/common/scripts/get-codeowners.ps1

```diff
@@ -1,49 +1,89 @@
+  dotnet tool install --tool-path $ToolPath --add-source $DevOpsFeed --version $ToolVersion "Azure.Sdk.Tools.RetrieveCodeOwners" | Out-Null
 
-foreach ($contentLine in $codeOwnersContent) {
-  if (-not $contentLine.StartsWith("#") -and $contentLine){
-    $splitLine = $contentLine -split "\s+"
-    
-    # CODEOWNERS file can also have labels present after the owner aliases
-    # gh aliases start with @ in codeowners. don't pass on to API calls
-    $ownedFolders[$splitLine[0].ToLower().Trim("/")] = ($splitLine[1..$($splitLine.Length)] `
-      | ? { $_.StartsWith("@") } `
-      | % { return $_.substring(1) }) -join ","
+  # Test to see if the tool properly installed.
+  if (!(Get-Command $command -errorAction SilentlyContinue)) {
+    Write-Error "The retrieve-codeowners tool is not properly installed. Please check your tool path. $ToolPath"
+    return 
   }
+  return $command
 }
 
-$results = $ownedFolders[$target]
-
-if ($results) {
-  Write-Host "Found a folder $results to match $target"
+function Get-CodeOwners ([string]$targetDirectory = $TargetDirectory, [string]$codeOwnerFileLocation = $CodeOwnerFileLocation)
+{
+  $command = Get-CodeOwnersTool
+  # Params $RootDirectory is already in use in cpp release pipeline. 
+  # Will use $codeOwnerFileLocation and deprecate $RootDirectory once it is ready to retire $RootDirectory.
+  if ($RootDirectory -and !(Test-Path $codeOwnerFileLocation)) {
+    $codeOwnerFileLocation = Join-Path $RootDirectory ".github/CODEOWNERS"
+  }
+  
+  $codeOwnersString = & $command --target-directory $targetDirectory --code-owner-file-path $codeOwnerFileLocation
+  # Failed at the command of fetching code owners.
+  if ($LASTEXITCODE -ne 0) {
+    return ,@()
+  }
+  
+  $codeOwnersJson = $codeOwnersString | ConvertFrom-Json
+  if (!$codeOwnersJson) {
+    Write-Host "No code owners returned from the path: $targetDirectory"
+    return ,@()
+  }
   
   if ($VsoVariable) {
-    $alreadyPresent = [System.Environment]::GetEnvironmentVariable($VsoVariable)
+    $codeOwners = $codeOwnersJson.Owners -join ","
+    Write-Host "##vso[task.setvariable variable=$VsoVariable;]$codeOwners"
+  }
+
+  return $codeOwnersJson.Owners
+}
 
-    if ($alreadyPresent) { 
-      $results += ",$alreadyPresent"
+function TestGetCodeOwner([string]$targetDirectory, [string]$codeOwnerFileLocation, [string[]]$expectReturn) {
+  $actualReturn = Get-CodeOwners -targetDirectory "sdk" -codeOwnerFileLocation $testFile
+
+  if ($actualReturn.Length -ne $expectReturn.Length) {
+    Write-Error "The length of actual result is not as expected. Expected length: $expectReturn.Length, Actual length: $actualReturn.Length."
+    exit 1
+  }
+  for ($i = 0; $i -lt $expectReturn.Length; $i++) {
+    if ($expectReturn[$i] -ne $actualReturn[$i]) {
+      Write-Error "Expect result $expectReturn[$i] is different than actual result $actualReturn[$i]."
+      exit 1
     }
-    Write-Host "##vso[task.setvariable variable=$VsoVariable;]$results"
   }
+  exit 0
```

Comment:
Having this exit 0 will cause the first test case to stop the others from running.